pkg/controllers/nodeclaim/lifecycle/launch.go

```diff
@@ -127,7 +129,14 @@ func PopulateNodeClaimDetails(nodeClaim, retrieved *v1.NodeClaim) *v1.NodeClaim
 		scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...).Labels(), // Single-value requirement resolved labels
 		nodeClaim.Labels, // User-defined labels
 	)
-	nodeClaim.Annotations = lo.Assign(nodeClaim.Annotations, retrieved.Annotations)
+
+	kubeletLabels := scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...).KubeletLabels()
+	json, err := json.Marshal(kubeletLabels)
+	if err != nil {
+		panic(err)
+	}
+	kubeletLabelsAnnotation := map[string]string{apis.Group + "/kubelet-labels": string(json)}
```

Comment:
```suggestion
	kubeletLabelsAnnotation := map[string]string{apis.Group + "/node-labels": string(json)}
```
We shouldn't assume that the node runs a kubelet. Any conformant node implementation would be OK.

Reply:
makes sense, I feel node is still conflating things though. How about something like
```
// nodeRestrictedLabelsAnnotation contains a list of labels created from the requirements but filtering out those rejected by the node restriction admission
nodeRestrictedLabelsAnnotation apis:= .Group + /node-restricted-labels
```
?